test/Guardian.t.sol

```diff
@@ -0,0 +1,367 @@
+        vm.stopPrank();
+
+        // Then: the user can supply
+        vm.startPrank(user);
+        lendingPool.depositGuarded(100);
+        vm.stopPrank();
+
+        // Then: the total supply is updated
+        assertEq(lendingPool.totalSupply(), 100);
+    }
+
+    function testSuccess_unPause_onlyUnpausePossible(uint256 timePassedAfterPause) public {
+        vm.assume(timePassedAfterPause <= 30 days);
+
+        // Given: the lending pool is paused
+        vm.startPrank(guardian);
+        lendingPool.pause();
+        vm.stopPrank();
+
+        // Given: Sometime passed after the pause
+        vm.warp(block.timestamp + timePassedAfterPause);
+
+        // When: the owner unPauses the supply
+        vm.startPrank(owner);
+        lendingPool.unPause(true, true, true, false, true);
+        vm.stopPrank();
+
+        // When: the owner attempts the pause the supply from the unPause
+        vm.startPrank(owner);
+        lendingPool.unPause(true, true, true, true, true);
+        vm.stopPrank();
+
+        // Then: the user can still supply because the once the supply is unPaused, it cannot be paused
+        vm.startPrank(user);
+        lendingPool.depositGuarded(100);
+        vm.stopPrank();
+
+        // Then: the total supply is updated
+        assertEq(lendingPool.totalSupply(), 100);
+    }
+
+    function testRevert_pause_timeNotExpired(uint256 timePassedAfterPause) public {
+        vm.assume(timePassedAfterPause < 32 days);
+
+        // Given: the lending pool is paused
+        vm.startPrank(guardian);
+        lendingPool.pause();
+        vm.stopPrank();
+
+        // Given: 1 day passed
+        uint256 startTimestamp = block.timestamp;
+        vm.warp(startTimestamp + 1 days);
+
+        // When: the owner unPauses
+        vm.startPrank(owner);
+        lendingPool.unPause(false, false, false, false, false);
+        vm.stopPrank();
+
+        // Then: the guardian cannot pause again until 32 days passed from the first pause
+        vm.warp(startTimestamp + timePassedAfterPause);
+        vm.expectRevert("Guardian: Cannot pause, Pause time not expired");
+        vm.startPrank(guardian);
+        lendingPool.pause();
+        vm.stopPrank();
+    }
+
+    function testRevert_unPause_userCannotUnPauseBefore30Days(uint256 timePassedAfterPause) public {
+        vm.assume(timePassedAfterPause < 30 days);
+
+        // Given: the lending pool is paused
+        vm.startPrank(guardian);
+        lendingPool.pause();
+        vm.stopPrank();
+
+        // Given: Sometime passed after the pause
+        vm.warp(block.timestamp + timePassedAfterPause);
+
+        // When: the user tries to unPause
+        vm.expectRevert("Guardian: Cannot unPause, unPause time not expired");
+        vm.startPrank(user);
+        lendingPool.unPause();
+        vm.stopPrank();
+    }
+
+    function testSuccess_unPause_userCanUnPauseAfter30Days(uint256 deltaTimePassedAfterPause) public {
```

Comment:
might need some changes if we will only allow users to unpause during day 30 to day 32th

Reply:
We allow users to unPause after 30th day after pause happened. 
We allow guardian to pause after 32nd day of previous pause happened. 

They are mutually compatible. So if guardian does not trigger pause again at 32th day. Users should still be able to unPause